### PR TITLE
yazi: init at 0.1.3

### PR DESCRIPTION
**Pull request by XYenon**
## Description of changes

[Yazi - ⚡️ Blazing Fast Terminal File Manager](https://github.com/sxyazi/yazi)

https://github.com/NixOS/nixpkgs/issues/250564

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/file-managers/yazi/default.nix [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  version = "0.1.3";
+
+  src = fetchFromGitHub {
+    owner = "sxyazi";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-IUE2846AltYyEefkavCJOEak9mW0wygojWsucwUEgh4=";
+  };
+
+  postPatch =
```
Comment (matthiasbeyer) [inline pkgs/applications/file-managers/yazi/default.nix:41]: I see what your intention was here, but that's not the way it should be done IMO!

I think what you should do is something like this:

```nix
installPhase = let
  runtimePaths = []
  ++ (lib.optionals withFile [ file ])
  # ... others accordingly
  ;
in
''
    runHook preInstall

    wrapProgram $out/bin/yazi \
        --prefix PATH : "${lib.makeBinPath runtimePaths}"

    runHook postInstall
'';

(above is untested).

Can you check whether this works as expected and update your PR?

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]

**Comment by teto** [comment]
looks pretty cool, I hoped this one would be able to load/display /nix/store async (ranger and joshuto blocks for a long time when listing /nix/store) but this isn't the one yet. It loaded /nix/store much faster than the other two (mine is about 200Gb I think)

**Comment by teto** [comment]
Result of `nixpkgs-review pr 250697` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>yazi</li>
  </ul>
</details>

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]
I like what I see! Could you squash the two commits and also put me into the maintainers please?
All in one commit please!

**Comment by XYenon** [comment]
> I like what I see! Could you squash the two commits and also put me into the maintainers please? All in one commit please!

Done :)

**Review by matthiasbeyer — APPROVED** [review]

**Comment by teto** [comment]
https://github.com/sxyazi/yazi#usage recommands to copy yazi default configuration files but these were not packaged ?

**Comment by XYenon** [comment]
> https://github.com/sxyazi/yazi#usage recommands to copy yazi default configuration files but these were not packaged ?

If you use the default config, copy is not necessary. And I well add a home manager module soon, which is already in [my NUR](https://github.com/XYenon/nur-packages/blob/master/hm-modules/programs/yazi.nix).

**Comment by teto** [comment]
nice ! you can ping me when opening the HM PR.

**Comment by aspauldingcode** [comment]
Why is yazi gone from nixpkgs?

**Comment by matthiasbeyer** [comment]
What do you mean with "gone"?

**Comment by aspauldingcode** [comment]
> What do you mean with "gone"?

Well it was gone that day. It's definitely back.. weird.

**Pull request merged**